src/main/java/seedu/address/logic/commands/quizcommands/CorrectCommand.java

```diff
@@ -27,11 +27,15 @@ public CommandResult execute(QuizModel model) throws CommandException {
         try {
             model.tallyScore(true);
             Question nextQuestion = model.getQuestion();
+
             QuizCommand.setStatus(Status.ON_QUESTION);
+            QuizCommand.updateCommandResult(nextQuestion.toString());
+
             return new CommandResult(nextQuestion.toString());
 
-        } catch (NullPointerException e) {
-            return new CommandResult("Score: " + model.stopQuiz());
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
+            QuizCommand.updateCommandResult(null);
+            return new CommandResult(model.stopQuiz());
```

Comment:
Can you slightly explain this part? Does the Quiz stop by running one more CorrectCommand and causes NullPointerException or IndexOutOfBoundsException? What is the scenario that the exceptions are thrown?

Thanks a lot~~

Reply:
Hey no problem! Each command leads to the next stage of the quiz, like `FlipCommand` / `AnswerCommand` would show the answer to the current question and increment the index of flashcards, while `CorrectCommand`/ `WrongCommand` would show the next question and check if the index has exceeded after the execution of the previous 2 types of commands. So the `CorrectCommand`/ `WrongCommand` is only run once per flashcard iterated.

The two exceptions will be thrown if the quiz reaches the end of the flashcard set, and the index being kept track of exceeds the number of flashcards.

Reply:
OHH, just too clarify, so the flow of usage is like,
quiz flset:1
=====> show the question
ans: my answer
=====> show the answer
correctCommand/wrongCommand
======> show next question or end the quiz if all flashcards have been iterated.

Am I correct?

Reply:
Yup, that is exactly right haha